packages/backend/src/models/HuggingFaceModelHandler.spec.ts

```diff
@@ -0,0 +1,157 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { EventEmitter } from '@podman-desktop/api';
+import type { TelemetryLogger } from '@podman-desktop/api';
+import { beforeEach, expect, test, vi } from 'vitest';
+import { ModelsManager } from '../managers/modelsManager';
+import type { CatalogManager } from '../managers/catalogManager';
+import type { ModelInfo } from '@shared/models/IModelInfo';
+import { TaskRegistry } from '../registries/TaskRegistry';
+import type { CancellationTokenRegistry } from '../registries/CancellationTokenRegistry';
+import type { PodmanConnection } from '../managers/podmanConnection';
+import type { ConfigurationRegistry } from '../registries/ConfigurationRegistry';
+import { ModelHandlerRegistry } from '../registries/ModelHandlerRegistry';
+import { HuggingFaceModelHandler } from './HuggingFaceModelHandler';
+import { snapshotDownload } from '@huggingface/hub';
+import type { RpcExtension } from '@shared/messages/MessageProxy';
+
+vi.mock('@podman-desktop/api', () => {
+  return {
+    EventEmitter: vi.fn(),
+  };
+});
+
+vi.mock('@huggingface/hub', () => {
+  return {
+    scanCacheDir: vi.fn(),
+    snapshotDownload: vi.fn(),
+  };
+});
+
+const rpcExtensionMock = {
+  fire: vi.fn(),
+} as unknown as RpcExtension;
+
+const catalogManagerMock = {
+  getModels(): ModelInfo[] {
+    return [
+      { id: 'model-id-1', name: 'model-id-1-model' } as ModelInfo,
+      { id: 'model-id-2', name: 'model-id-2-model' } as ModelInfo,
+    ];
+  },
+  onUpdate: vi.fn(),
+} as unknown as CatalogManager;
+
+const telemetryLogger = {
+  logUsage: vi.fn(),
+  logError: vi.fn(),
+} as unknown as TelemetryLogger;
+
+const taskRegistry: TaskRegistry = new TaskRegistry(rpcExtensionMock);
+
+const cancellationTokenRegistryMock = {
+  createCancellationTokenSource: vi.fn(),
+} as unknown as CancellationTokenRegistry;
+
+const podmanConnectionMock = {
+  getContainerProviderConnections: vi.fn(),
+} as unknown as PodmanConnection;
+
+const configurationRegistryMock = {
+  getExtensionConfiguration: vi.fn(),
+} as unknown as ConfigurationRegistry;
+
+const modelHandlerRegistry = new ModelHandlerRegistry(rpcExtensionMock);
+
+const modelsManager: ModelsManager = new ModelsManager(
+  rpcExtensionMock,
+  catalogManagerMock,
+  telemetryLogger,
+  taskRegistry,
+  cancellationTokenRegistryMock,
+  podmanConnectionMock,
+  configurationRegistryMock,
+  modelHandlerRegistry,
+);
+
+const huggingFaceModelHandler = new HuggingFaceModelHandler(modelsManager);
+
+beforeEach(() => {
+  const listeners: ((value: unknown) => void)[] = [];
+
+  vi.mocked(EventEmitter).mockReturnValue({
+    event: vi.fn().mockImplementation(callback => {
+      listeners.push(callback);
+    }),
+    fire: vi.fn().mockImplementation((content: unknown) => {
+      listeners.forEach(listener => listener(content));
+    }),
+  } as unknown as EventEmitter<unknown>);
+});
```

Comment:
not a blocker,
using vi.fn().mockImplementation loses the typing.

You may do instead smth like:
```
const eventReturned = {
  event: vi.fn(),
  fire: vi.fn(),
};

vi.mocked(EventEmitter).mockReturnValue(eventReturned);

vi.mocked(eventReturned.event).mockImplementation(...);
vi.mocked(eventReturned.fire).mockImplementation(...);
```

Reply:
I think I copied it from elsewhere but will update